llvm/lib/CodeGen/SelectionDAG/DAGCombiner.cpp

```diff
@@ -5236,6 +5236,23 @@ SDValue DAGCombiner::visitAVG(SDNode *N) {
     return DAG.getNode(ISD::SRL, DL, VT, X,
                        DAG.getShiftAmountConstant(1, VT, DL));
 
+  // fold avgu(zext(x), zext(y)) -> zext(avgu(x, y))
+  SDValue A;
+  SDValue B;
+  if (sd_match(N, m_c_BinOp(ISD::AVGFLOORU, m_ZExt(m_Value(A)),
```

Comment:
```suggestion
  if (sd_match(N, m_BinOp(ISD::AVGFLOORU, m_ZExt(m_Value(A)),
```

Reply:
AVGFLOORU/AVGCEILU are commutative so m_c_BinOp should work no?

Reply:
We don't need commutative matchers as both LHS and RHS are the same.